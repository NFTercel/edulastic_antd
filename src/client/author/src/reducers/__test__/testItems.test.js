import testItemsReducer from "../testItems";

import {
  RECEIVE_TEST_ITEMS_REQUEST,
  RECEIVE_TEST_ITEMS_SUCCESS,
  RECEIVE_TEST_ITEMS_ERROR,
  SET_TEST_ITEMS_REQUEST
} from "../../constants/actions";

describe("test items", () => {
  const initialState = {
    items: [],
    error: null,
    loading: false,
    page: 1,
    limit: 5,
    count: 0
  };

  it("should return the initial state", () => {
    expect(testItemsReducer(initialState, {})).toMatchSnapshot();
  });

  it("should return the receive test items request state", () => {
    expect(
      testItemsReducer(initialState, {
        type: RECEIVE_TEST_ITEMS_REQUEST,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it("should return the receive test items success state", () => {
    expect(
      testItemsReducer(initialState, {
        type: RECEIVE_TEST_ITEMS_SUCCESS,
        payload: {
          items: [
            {
              rows: [
                {
                  tabs: [],
                  dimension: "100%",
                  widgets: [
                    {
                      widgetType: "question",
                      type: "multipleChoice",
                      title: "Multiple choice",
                      reference: "5c0f0c9ed8c7f4453f5bb51f",
                      tabIndex: 0
                    }
                  ]
                }
              ],
              columns: []
            }
          ],
          count: 2
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the receive test items error state", () => {
    expect(
      testItemsReducer(initialState, {
        type: RECEIVE_TEST_ITEMS_ERROR,
        payload: {
          error: "unexpected error happened when receive test items"
        }
      })
    ).toMatchSnapshot();
  });

  expect(
    testItemsReducer(initialState, {
      type: SET_TEST_ITEMS_REQUEST,
      payload: {}
    })
  ).toMatchSnapshot();
});
