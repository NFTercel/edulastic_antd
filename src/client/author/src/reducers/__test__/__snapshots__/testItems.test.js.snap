// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "count": 0,
  "error": null,
  "items": Array [],
  "limit": 5,
  "loading": false,
  "page": 1,
  "selectedItems": Object {},
}
`;

exports[`test items should return the initial state 1`] = `
Object {
  "count": 0,
  "error": null,
  "items": Array [],
  "limit": 5,
  "loading": false,
  "page": 1,
}
`;

exports[`test items should return the receive test items error state 1`] = `
Object {
  "count": 0,
  "error": "unexpected error happened when receive test items",
  "items": Array [],
  "limit": 5,
  "loading": false,
  "page": 1,
}
`;

exports[`test items should return the receive test items request state 1`] = `
Object {
  "count": 0,
  "error": null,
  "items": Array [],
  "limit": 5,
  "loading": true,
  "page": 1,
}
`;

exports[`test items should return the receive test items success state 1`] = `
Object {
  "count": 2,
  "error": null,
  "items": Array [
    Object {
      "columns": Array [],
      "rows": Array [
        Object {
          "dimension": "100%",
          "tabs": Array [],
          "widgets": Array [
            Object {
              "reference": "5c0f0c9ed8c7f4453f5bb51f",
              "tabIndex": 0,
              "title": "Multiple choice",
              "type": "multipleChoice",
              "widgetType": "question",
            },
          ],
        },
      ],
    },
  ],
  "limit": 5,
  "loading": false,
  "page": 1,
}
`;
